cache/cache.go

```diff
@@ -72,13 +75,27 @@ func New(watcher clientv3.Watcher, prefix string, opts ...Option) (*Cache, error
 		prefix:      prefix,
 		cfg:         cfg,
 		watcher:     watcher,
+		store:       newStore(),
 		ready:       make(chan struct{}),
 		stop:        cancel,
 		internalCtx: internalCtx,
 	}
 
 	cache.demux = NewDemux(internalCtx, &cache.waitGroup, cfg.HistoryWindowSize, cfg.ResyncInterval)
 
+	var startRev int64
+	if kv, ok := watcher.(clientv3.KV); ok {
```

Comment:
This block should not be called once on cache start, but every time watch is broken and